staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -518,6 +518,14 @@ func InClusterConfig() (*Config, error) {
 	if len(host) == 0 || len(port) == 0 {
 		return nil, ErrNotInCluster
 	}
+	// check whether tokenFile exists
+	if _, err := os.Stat(tokenFile); os.IsNotExist(err) {
+		return nil, ErrNotInCluster
```

Comment:
the token file may not exist if automountServiceAccountToken is set to false on the pod or service account... it would be misleading to return `ErrNotInCluster` in that case